src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -626,6 +626,18 @@ private static bool SymbolIsAnnotatedAsPreview(ISymbol symbol, ConcurrentDiction
                     return true;
                 }
 
+                if (symbol.ContainingAssembly != null && symbol.ContainingAssembly.HasAttribute(previewFeatureAttribute))
+                {
+                    requiresPreviewFeaturesSymbols.GetOrAdd(symbol, true);
+                    return true;
+                }
+
+                if (symbol.ContainingModule != null && symbol.ContainingModule.HasAttribute(previewFeatureAttribute))
+                {
+                    requiresPreviewFeaturesSymbols.GetOrAdd(symbol, true);
+                    return true;
+                }
+
                 INamedTypeSymbol? parent = symbol.ContainingType;
                 if (parent is INamespaceSymbol)
                 {
```

Comment:
If it needs to be `symbol.ContainingType` please remove this check, i don't think containing type would go up to namespace

Reply:
Changing to ContainingSymbol, so keeping this block